crates/storage/db/src/abstraction/common.rs

```diff
@@ -16,13 +18,20 @@ pub type IterPairResult<T> = Option<Result<KeyValue<T>, DatabaseError>>;
 /// A value only result for table `T`.
 pub type ValueOnlyResult<T> = Result<Option<<T as Table>::Value>, DatabaseError>;
 
-use crate::{abstraction::table::*, DatabaseError};
-
-// Sealed trait helper to prevent misuse of the API.
+// Sealed trait helper to prevent misuse of the Database API.
 mod sealed {
+    use crate::{database::Database, mock::DatabaseMock, DatabaseEnv};
+    use std::sync::Arc;
+
+    /// Sealed trait to limit the implementors of the Database trait.
     pub trait Sealed: Sized {}
-    #[allow(missing_debug_implementations)]
-    pub struct Bounds<T>(T);
-    impl<T> Sealed for Bounds<T> {}
+
+    impl<DB: Database> Sealed for &DB {}
+    impl<DB: Database> Sealed for Arc<DB> {}
+    impl Sealed for DatabaseEnv {}
+    impl Sealed for DatabaseMock {}
```

Comment:
any good reason to keep this?

Reply:
just to retain total control, but can be removed imo